### ISSUE TITLE
[Header] "text alignement" should display accordingly when there is an icon as content

### ISSUE THREAD

**Issue by challet** [Semantic-Org/Semantic-UI#6902]
### Steps

* Display a header with a [text alignement](https://semantic-ui.com/elements/header#text-alignment), for instance `ui centered aligned header`
* Add an [icon](https://semantic-ui.com/elements/header#icon) and a `<div ="content">` (as seen in documentation example) in the header.

### Expected Result

The header content is aligned accordingly.

### Actual Result

The header content stays left aligned.

### Version
2.4.2

### Testcase
https://jsfiddle.net/w236rdmz/

* it works as expected when there is no `<div class="content">` following the icon 
* removing the *display: table-cell* property on both the `<i>` and `<div>` elements seems to fix it, though I guess it might break other cases.

**Comment by VanquishedWombat** [comment]
Maybe not an answer but the HTML spec informs us that H2 is meant to contain only 'phrasing content' and not structural stuff like div's.  See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements. Since SUI is building on HTML it is usually worth following HTML principles as a foundation and SUI then applies pleasing CSS.

Re-writing your code, removing the div inside the H2 seems to work, though vertical alignment may need attention.

<div class="ui left centered header">    
    <div class="content">
      <h2>
        <i aria-hidden="true" class="large user icon"></i> has a div.content
      </h2>
  </div>
</div>

**Comment by challet** [comment]
That's right, so seeing it the other way around and replacing the divs by spans, wich belongs to [phrasing content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content), the problem remains : https://jsfiddle.net/sv4whxoq/

Also added a case with only `<div>`s as I thought it was in the documentation example, but it happens that the "icon" one is using `<h2>` as well : https://semantic-ui.com/elements/header.html#icon
The one about "text alignment" has indeed no div inside. It confuses me, but imho variations should be compatible with all kind of types and contents referenced.
